src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogCollectorProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Loggers;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    internal class FunctionInstanceLogCollectorProvider : IEventCollectorProvider
+    {
+        private readonly IFunctionMetadataManager _metadataManager;
+        private readonly IProxyMetadataManager _proxyMetadataManager;
+        private readonly IMetricsLogger _metrics;
+        private readonly IHostIdProvider _hostIdProvider;
+        private readonly IConfiguration _configuration;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public FunctionInstanceLogCollectorProvider(IFunctionMetadataManager metadataManager, IProxyMetadataManager proxyMetadataManager,
+            IMetricsLogger metrics, IHostIdProvider hostIdProvider, IConfiguration configuration, ILoggerFactory loggerFactory)
+        {
```

Comment:
small nit: `null` validation on those things. Been trying to make sure validation happens as early as possible as it helps us narrow down the source of issues.

Reply:
Yeah good point. Haven't merged yet -- I'll get that in.